feature_engine/discretisation/base_discretiser.py

```diff
@@ -0,0 +1,120 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import pandas as pd
+
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.dataframe_checks import _check_contains_na
+
+import warnings
+
+
+class BaseDiscretiser(BaseNumericalTransformer):
+    """
+    Share set-up checks and methods across numerical discretizers
+
+    Parameters
+    ----------
+    return_object: bool, default=False
+        Whether the the discrete variable should be returned as numeric or as
+        object. If you would like to proceed with the engineering of the variable as if
+        it was categorical, use True. Alternatively, keep the default to False.
+
+    return_boundaries : bool, default=False
+        Whether the output should be the interval boundaries. If True, it returns
+        the interval boundaries. If False, it returns integers.
+
+
+    """
+
+    def __init__(
+            self,
+            return_object: bool = False,
+            return_boundaries: bool = False,
+    ) -> None:
+
+        if not isinstance(return_object, bool):
+            raise ValueError("return_object must be True or False")
```

Comment:
Since we are here, maybe we can expand the error messages to inform the user what value they actually entered. Can you modify the error messages so they have the sentence: `f"Got {entered_parameter} instead."`

So for this example, it should look like this:

```
raise ValueError("return_object must be True or False. "
                   f"Got {return_object} instead.)
```

Thank you!

Some nitpicks: be mindful of the . and the space after it. Otherwise, the second sentence will be printed out just next to the dot in the users console.